### ISSUE TITLE
how to change Predict.py to test only from one image

### ISSUE THREAD

**Issue by mohitjain1943** [tuanphan09/captcha-recognition#8]
HI @tuanphan09 ,

I want to run Predict.py file as an http service so i just want to know how to change predict.py file to run for 1 image like i pass 1 image it will predict the text of that image. Please help me with that i am now in machine learning so its getting difficult to change file.

**Comment by tuanphan09** [comment]
You can find the answer in this comment
https://github.com/tuanphan09/captcha-recognition/issues/5#issuecomment-548689954

**Comment by mohitjain1943** [comment]
HI @tuanphan09 

I Used below code 

`import os
import itertools
import numpy as np
import pandas as pd
import operator
import math
from sklearn.model_selection import train_test_split
import time
from keras import backend as K
from keras.callbacks import *
from keras.layers import *
from keras.models import *
from keras.utils import *
from keras.optimizers import Adadelta, RMSprop, Adam, SGD
from keras.callbacks import ModelCheckpoint
from keras.callbacks import TensorBoard
import cv2 

from config import *
from model import *

def greedy_search_decoder(data):
    return np.argmax(data, axis=1)

def get_label(idxes):
    result = ''
    for i in range(len(idxes)):
            if (i == 0 or idxes[i] != idxes[i-1]) and idxes[i] < len(characters):
                    result += characters[idxes[i]]
    return result

def decode_label_greedy(data):
    idxes = greedy_search_decoder(data)
    return get_label(idxes)


# load model
base_model = CRNN_model(is_training=False)
base_model.load_weights(base_model_path)
print("Done loaded model!")

img_path = 'C:/Users/mohit/Downloads/captcha-recognition-master/data/public_test/0a4dfab63233f86fcd77e3c8647d2adf.png'
img = cv2.imread(img_path, cv2.IMREAD_GRAYSCALE)  # (height, width)
img = np.reshape(img, (*img.shape, 1))  # (height, width, 1)
img = img.transpose([1, 0, 2])    # (width, height, 1)

X = np.array([img], dtype=np.float32)  
X /= 255  # normalize

y_pred = base_model.predict(X)  # X.shape = (number of images, width, height, 1)

text = decode_label_greedy(y_pred[:, 2:])
print(text)`

but i am getting attached error please help me

![Capture](https://user-images.githubusercontent.com/6769471/81138879-36bb0200-8f81-11ea-8eb4-b02c5f7c6437.PNG)

I am using 
windows 10
python 3.6.6
tensorflow-gpu 1.5.0
Keras 2.3.1
Keras-Applications 1.0.8
keras-ocr 0.8.3
Keras-Preprocessing 1.1.0

**Comment by tuanphan09** [comment]
Sorry, you can change this line:
```
text = decode_label_greedy(y_pred[:, 2:])
```
into
```
text = decode_label_greedy(y_pred[0, 2:])
```

and it'll work well.

**Comment by mohitjain1943** [comment]
wow thax @tuanphan09 .....worked liked charms...your help is really appreciated.
